packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1963,7 +1958,12 @@ function createChildReconciler(
       // the error or suspending if needed.
       const throwFiber = createFiberFromThrow(x, returnFiber.mode, lanes);
       throwFiber.return = returnFiber;
+      if (__DEV__) {
+        throwFiber._debugInfo = currentDebugInfo;
```

Comment:
This is the effective change. We now can pick this up from module scope before clearing it.